File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.java
class Solution {
    public int minimumArea(int[][] grid) {
        
        int minRow=Integer.MAX_VALUE;
        int maxRow=Integer.MIN_VALUE;
        int minCol=Integer.MAX_VALUE;
        int maxCol=Integer.MIN_VALUE;

        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(grid[i][j]==1){
                    minRow=Math.min(minRow,i);
                    maxRow=Math.max(maxRow,i);
                    minCol=Math.min(minCol,j);
                    maxCol=Math.max(maxCol,j);
                }
            }
        }
        if(minRow==Integer.MAX_VALUE) {
            return 0;
        }
        return (maxRow-minRow+1)*(maxCol-minCol+1);
    }
}

File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var minimumArea = function(grid) {
    
    let minCol = grid[0].length, maxCol = 0, minRow = grid.length, maxRow = 0;

    for(let i = 0; i < grid.length; i++) {

        for(let j = 0; j < grid[0].length; j++) {

            if(grid[i][j] === 0) continue;

            minCol = Math.min(minCol, j);
            maxCol = Math.max(maxCol, j);
            minRow = Math.min(minRow, i);
            maxRow = Math.max(maxRow, i);
        }
    }

    return (maxCol - minCol + 1) * (maxRow - minRow + 1);
};